includes/blocks/class-sensei-course-outline-block.php

```diff
@@ -238,4 +236,82 @@ public function render_course_outline_block( $attributes ) {
 
 	}
 
+	/**
+	 * Initialize notices.
+	 *
+	 * @internal
+	 *
+	 * @since $$next-version$$
+	 */
+	public function frontend_notices() {
+		$post = get_post();
+
+		if ( ! $post || 'course' !== $post->post_type || is_admin() ) {
+			return;
+		}
+
+		$course_id        = $post->ID;
+		$structure        = Sensei_Course_Structure::instance( $course_id )->get( 'view' );
+		$has_draft        = $this->has_draft( $structure );
+		$edit_cta_message = '';
+
+		// Complete message with CTA if user can edit the course.
+		if ( Sensei_Course::can_current_user_edit_course( $course_id ) ) {
+			$edit_cta = sprintf(
+				'<a href="%s">%s</a>',
+				esc_url( get_edit_post_link( $course_id ) ),
+				__( 'your lessons', 'sensei-lms' )
+			);
+
+			$edit_cta_message = sprintf(
+				/* translators: %s: Edit course link. */
+				__( "When you're ready, let's publish %s in order to make them available to your students.", 'sensei-lms' ),
+				$edit_cta
+			);
+		}
+
+		// Notice for empty structure. Notice that draft lessons don't return for students, so it's considered empty.
+		if ( empty( $structure ) ) {
+			$message = __( 'There are no published lessons in this course yet.', 'sensei-lms' );
```

Comment:
As commented in https://github.com/Automattic/sensei/issues/7374#issuecomment-1867760211, I kept this message for students.
It's also displayed for the admin if there are no lessons in the course.